WordPress/openverse-frontend
# Re-fetch single item if it was taken from search results or related media

## Fixes


Fixes #1708 by @obulat

## Description


On opening a single result page, we re-use the data that we already have in the media/index store (if a search page link is clicked) or the related media store (if a related media link is clicked). This makes the loading of the single result fast because we already have the image thumbnail, title, attribution, etc. However, those results do not contain some important items such as album, file type, and file size.

This PR loads the single result page with the available information and fetches the single result endpoint from the API. When that fetch is ready, the single result page is updated with the missing information.

## Testing Instructions

Try loading some audios that are part of albums from the search or related media section. They all should load the album information. Try reloading the page - the server-rendered single detail page should have the same information.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
